### PR TITLE
[Enhancement] Add option to choose relative path of BE's debug symbol

### PR DESCRIPTION
**Pull request by zhangbutao**
## Why I'm doing:
https://github.com/StarRocks/starrocks/pull/46030 forced the be .c source file to use relative path.
But If using Clion to do the gdb remote debug, Clion always uses the absolute path & relative path will result in breakpoint failure.
sr_debug

## What I'm doing:
Provide a option to trun ON | OFF the relative path, so that users can use Clion to do gdb debug. default is `ON` , `ON `means  turning on the relative path.

If you want to turn off the relative path, you can build with `--with-source-file-relative-path OFF`
./build.sh  --be --with-source-file-relative-path OFF

Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.3
  - [x] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: be/CMakeLists.txt [hunk 1]
```diff
@@ -97,6 +97,8 @@ option(USE_AVX512 "Build with AVX512f/AVX512BW instruction" OFF)
 
 option(WITH_TENANN "Build with TENANN" OFF)
 
+option(WITH_RELATIVE_PATH "Build .c source file with relative path" OFF)
```
Comment (kevincai) [inline be/CMakeLists.txt:100]:
```suggestion
option(WITH_RELATIVE_SRC_PATH "Build source file with relative path" ON)
```

make it `ON` by default.

File: build.sh [hunk 2]
```diff
@@ -159,6 +162,7 @@ USE_STAROS=OFF
 BUILD_JAVA_EXT=ON
 OUTPUT_COMPILE_TIME=OFF
 WITH_TENANN=ON
+WITH_RELATIVE_PATH=OFF
```
Comment (kevincai) [inline build.sh:165]: make it `ON` by default in `build.sh`

change the `run-be-ut.sh`, make it `OFF` in that script.

File: be/CMakeLists.txt (hunk 1 above)
Comment (zhangbutao) [reply be/CMakeLists.txt]: Fixed.

File: build.sh (hunk 2 above)
Comment (zhangbutao) [reply build.sh]: Fixed.

File: build.sh [hunk 3]
```diff
@@ -292,6 +297,7 @@ echo "Get params:
     BUILD_JAVA_EXT              -- $BUILD_JAVA_EXT
     OUTPUT_COMPILE_TIME         -- $OUTPUT_COMPILE_TIME
     WITH_TENANN                 -- $WITH_TENANN
+    WITH_RELATIVE_SRC_PATH          -- $WITH_RELATIVE_SRC_PATH
```
Comment (kevincai) [inline build.sh:300]:
```suggestion
    WITH_RELATIVE_SRC_PATH       -- $WITH_RELATIVE_SRC_PATH
```

make it aligned with the previous line.

Comment (zhangbutao) [reply build.sh]: Done.

**Review by kevincai — DISMISSED** [review]

**Comment by zhangbutao** [comment]
Rebased to fix the code conflict.

File: build.sh [hunk 4]
```diff
@@ -430,7 +436,8 @@ if [ ${BUILD_BE} -eq 1 ] || [ ${BUILD_FORMAT_LIB} -eq 1 ] ; then
                   -DWITH_TENANN=${WITH_TENANN}                          \
                   -DSTARROCKS_JIT_ENABLE=${ENABLE_JIT}                  \
                   -DCMAKE_EXPORT_COMPILE_COMMANDS=ON  ..                \
-                  -DBUILD_FORMAT_LIB=${BUILD_FORMAT_LIB}
+                  -DBUILD_FORMAT_LIB=${BUILD_FORMAT_LIB}                \
+                  -DWITH_RELATIVE_SRC_PATH=${WITH_RELATIVE_SRC_PATH}
```
Comment (kevincai) [inline build.sh:440]: didn't notice that the `..` has been moved into middle. please correct it as follows

```suggestion
                  -DCMAKE_EXPORT_COMPILE_COMMANDS=ON                  \
                  -DBUILD_FORMAT_LIB=${BUILD_FORMAT_LIB}                \
                  -DWITH_RELATIVE_SRC_PATH=${WITH_RELATIVE_SRC_PATH} \
                  ..
```

Comment (zhangbutao) [reply build.sh]: Fixed.

**Review by kevincai — APPROVED** [review]

**Review by kevincai — APPROVED** [review]

**Review by wyb — APPROVED** [review]

**Pull request merged**